os.root("C:/") fails in Windows

The following script illustrates the problem:'

```scala
#!/usr/bin/env -S scala-cli shebang

//> using scala "3.3.1"
//> using lib "com.lihaoyi::os-lib::0.9.2"

import os._

val root1 = os.root("C:\\")
printf("cRoot: %s\n", root1)
val root2 = os.root("C:/")
printf("cRoot: %s\n", root2)
```

The output:
```scala
# /opt/ue/jsrc/oslibRootBug.sc
cRoot: C:\
Exception in thread "main" java.lang.AssertionError: assertion failed: C:/ is not a root path
        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
        at os.package$.root(package.scala:18)
        at oslibRootBug$_.<init>(oslibRootBug.sc:10)
        at oslibRootBug_sc$.script$lzyINIT1(oslibRootBug.sc:27)
        at oslibRootBug_sc$.script(oslibRootBug.sc:27)
        at oslibRootBug_sc$.main(oslibRootBug.sc:31)
        at oslibRootBug_sc.main(oslibRootBug.sc)
```

The fix and associated test case are here: #235

--- a/os/src-jvm/package.scala
+++ b/os/src-jvm/package.scala
@@ -15,7 +15,7 @@ package object os {
 
   def root(root: String, fileSystem: FileSystem = FileSystems.getDefault()): Path = {
     val path = Path(fileSystem.getPath(root))
-    assert(path.root == root, s"$root is not a root path")
+    assert(path.root == root || path.root == root.replace('/', '\\'), s"$root is not a root path")
     path
   }
 

--- a/os/src-native/package.scala
+++ b/os/src-native/package.scala
@@ -12,7 +12,7 @@ package object os {
 
   def root(root: String, fileSystem: FileSystem = FileSystems.getDefault()): Path = {
     val path = Path(fileSystem.getPath(root))
-    assert(path.root == root, s"$root is not a root path")
+    assert(path.root == root || path.root == root.replace('/', '\\'), s"$root is not a root path")
     path
   }
 

--- a/os/test/src-jvm/PathTestsCustomFilesystem.scala
+++ b/os/test/src-jvm/PathTestsCustomFilesystem.scala
@@ -226,8 +226,10 @@ object PathTestsCustomFilesystem extends TestSuite {
 
   val testWindows = Tests {
     test("cRootPath") {
-      val p = os.root("C:\\") / "Users"
-      assert(p.toString == "C:\\Users")
+      val p1 = os.root("C:\\") / "Users"
+      assert(p1.toString == "C:\\Users")
+      val p2 = os.root("C:/") / "Users"
+      assert(p2.toString == "C:\\Users")
     }
   }
 

